tools/common/Driver.cs

```diff
@@ -276,13 +295,19 @@ public static int RunCommand (string path, string args, string[] env = null, Str
 				stderr_completed.WaitOne (TimeSpan.FromSeconds (1));
 				stdout_completed.WaitOne (TimeSpan.FromSeconds (1));
 
+				output_received (null);
+
 				if (p.ExitCode != 0) {
 					// note: this repeat the failing command line. However we can't avoid this since we're often
 					// running commands in parallel (so the last one printed might not be the one failing)
-					if (!suppressPrintOnErrors)
-						Console.Error.WriteLine ("Process exited with code {0}, command:\n{1} {2}{3}", p.ExitCode, path, args, output.Length > 0 ? "\n" + output.ToString () : string.Empty);
+					if (!suppressPrintOnErrors) {
+						var msg = $"Process exited with code {p.ExitCode}, command:\n{path}";
+						if (output != null && output.Length > 0)
+							msg += $"\n{output}";
+						Console.Error.WriteLine (msg);
```

Comment:
There are other comments that explain this: we do things in parallel, and having multiple CWLs means it will be much more likely that other threads will write to stdout at the same time, interspersing their output with this output, and making a big incomprehensible mess out of it all.